tests/Countries/SwitzerlandTest.php

```diff
@@ -31,28 +67,127 @@
         ->not()->toBeEmpty();
 
     expect(formatDates($holidays))->toMatchSnapshot();
-});
+})->with('cantons');
 
 it('throws an error when an invalid region is given', function () {
-    new Switzerland('ch-xx');
+    $switzerland = new Switzerland('ch-xx');
+    $holidays = Holidays::for($switzerland)->get();
+
 })->throws(InvalidRegion::class);
 
-it('can translate swiss holidays into french', function () {
-    $holidays = Holidays::for(country: 'ch', locale: 'fr', year: 2024)->get();
+describe('supported languages', function () {
+    it('shares same translations keys in every language', function () {
 
-    expect($holidays)
-        ->toBeArray()
-        ->not()->toBeEmpty();
+        $filePaths = __DIR__.'/../../lang/switzerland/**/holidays.json';
+        $filePaths = glob($filePaths);
 
-    expect(formatDates($holidays))->toMatchSnapshot();
+        $keysArray = [];
+
+        foreach ($filePaths as $filePath) {
+            $content = file_get_contents($filePath);
+            $data = json_decode($content, true);
+            $keysArray[] = array_keys($data);
+        }
+
+        // Compare keys of the first JSON file with the others
+        for ($i = 1; $i < count($keysArray); $i++) {
+            $diff = array_diff($keysArray[0], $keysArray[$i]);
+            if (! empty($diff)) {
+
+                dump($filePaths[$i], $diff);
+            }
+
+            expect($diff)->toBeEmpty();
+        }
+
+    });
+
+    it('can translate swiss holidays into', function (string $language) {
+        $holidays = Holidays::for(country: 'ch', locale: $language)->get();
+
+        expect($holidays)
+            ->toBeArray()
+            ->not()->toBeEmpty();
+
+        expect(formatDates($holidays))->toMatchSnapshot();
```

Comment:
No need for a big snapshot tests here. You can just verify if a certain holiday name is present.